File: ViewModels/StudentDashboardViewModel.swift
//
//  StudentDashboardViewModel.swift
//  UDrive
//
//  Created on 2024
//

import Foundation

class StudentDashboardViewModel: ObservableObject {
    @Published var activeRides: [Ride] = []
    @Published var recentRides: [Ride] = []
    
    private let mockService = MockDataService.shared
    
    func loadRides(for userId: UUID) {
        activeRides = mockService.getActiveRides(for: userId)
        recentRides = mockService.rides.filter { $0.studentId == userId && $0.status == .completed }
            .sorted { $0.requestedAt > $1.requestedAt }
            .prefix(5)
            .map { $0 }
    }
}

File: lib/viewmodels/student_dashboard_viewmodel.dart
import 'package:flutter/foundation.dart';
import '../services/mock_data_service.dart';
import '../models/ride.dart';

class StudentDashboardViewModel extends ChangeNotifier {
  final MockDataService _mockService = MockDataService();
  List<Ride> _activeRides = [];
  List<Ride> _recentRides = [];

  List<Ride> get activeRides => List.unmodifiable(_activeRides);
  List<Ride> get recentRides => List.unmodifiable(_recentRides);

  void loadRides(String userId) {
    _activeRides = _mockService.getActiveRides(userId);
    _recentRides = _mockService.rides
        .where((ride) =>
            ride.studentId == userId && ride.status == RideStatus.completed)
        .toList();
    notifyListeners();
  }
}

